File: openapi/scalaz/src/main/scala/org/openapitools/client/api/MailContact.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import MailContact._

case class MailContact (
  /* The email address */
  email: String,
/* Optional contact name */
  name: Option[String])

object MailContact {
  import DateTimeCodecs._

  implicit val MailContactCodecJson: CodecJson[MailContact] = CodecJson.derive[MailContact]
  implicit val MailContactDecoder: EntityDecoder[MailContact] = jsonOf[MailContact]
  implicit val MailContactEncoder: EntityEncoder[MailContact] = jsonEncoderOf[MailContact]
}

File: openapi-client/javascript-closure-angular/API/Client/MailContact.js
goog.provide('API.Client.MailContact');

/**
 * An Email Contact
 * @record
 */
API.Client.MailContact = function() {}

/**
 * The email address
 * @type {!string}
 * @export
 */
API.Client.MailContact.prototype.email;

/**
 * Optional contact name
 * @type {!string}
 * @export
 */
API.Client.MailContact.prototype.name;

